algebird-core/src/main/scala/com/twitter/algebird/QTree.scala

```diff
@@ -108,32 +152,45 @@ class QTreeSemigroup[A](k: Int)(implicit val underlyingMonoid: Monoid[A]) extend
   }
 }
 
-case class QTree[A](
+case class QTree[@specialized(Int, Long, Float, Double, Unit) A] private[algebird] (
```

Comment:
I'm happy to do that, think it looks nicer than this use of private? we have a few apply's already, unapply we probably want to more match the old apply too. So it might wind up making some of it nicer to migrate to